src/OpenDebugAD7/TracepointManager.cs

```diff
@@ -0,0 +1,229 @@
+                            case "FILEPOS":
+                            case "FUNCTION":
+                            case "ADDRESS":
+                            case "TID":
+                            case "TNAME":
+                            case "PID":
+                            case "PNAME":
+                            case "CALLER":
+                            case "CALLSTACK":
+                            case "TICK":
+                                tokensUsed.Add(token);
+                                return InterpolateToken(token);
+                            default:
+                                return match.Value;
+                        }
+                    }
+                    else
+                    {
+                        string expression = match.Value.Substring(1, match.Value.Length - 2);
+                        try
+                        {
+                            return InterpolateVariable(expression, topFrame[0].m_pFrame, radix);
+                        }
+                        catch (AD7Exception e)
+                        {
+                            return e.ToString();
+                        }
+                    }
+                }
+                else
+                {
+                    return string.Empty;
+                }
+            });
+
+            Dictionary<string, object> eventProperties = null;
+            if (tokensUsed.Count > 0)
+            {
+                eventProperties = new Dictionary<string, object>();
+                eventProperties.Add(DebuggerTelemetry.TelemetryTracepointTokens, tokensUsed);
+            }
+            DebuggerTelemetry.ReportEvent(DebuggerTelemetry.TelemetryTracepointEventName, eventProperties);
+
+            return interpolatedLogMessage;
+        }
+
+        private string InterpolateToken(string token)
+        {
+            return string.Format(CultureInfo.CurrentCulture, "<Not Implemented: ${0}>", token);
+        }
+
+        private string InterpolateVariable(string variable, IDebugStackFrame2 topFrame, uint radix)
+        {
+            int hr = HRConstants.S_OK;
+            string errorMessage = string.Format(CultureInfo.CurrentCulture, "<Evaluation Error: {{{0}}}>", variable);
+            ErrorBuilder eb = new ErrorBuilder(() => errorMessage);
+
+            IDebugExpressionContext2 expressionContext;
+            hr = topFrame.GetExpressionContext(out expressionContext);
+            eb.CheckHR(hr);
+
+            IDebugExpression2 expressionObject;
+            hr = expressionContext.ParseText(variable, enum_PARSEFLAGS.PARSE_EXPRESSION, radix, out expressionObject, out string _, out uint _);
+            eb.CheckHR(hr);
+            eb.CheckOutput(expressionObject);
+
+            IDebugProperty2 property;
+            if (expressionObject is IDebugExpressionDAP expressionDapObject)
+            {
+                DAPEvalFlags dapEvalFlags = DAPEvalFlags.CLIPBOARD_CONTEXT; // Use full string for tracepoints.
```

Comment:
>DAPEvalFlags dapEvalFlags = DAPEvalFlags.CLIPBOARD_CONTEXT; // Use full string for tracepoints. [](start = 16, length = 95)

Does GDB have particularly small strings? I am a bit skeptical that you want gigantic strings here.

Reply:
The default size GDB prints is 200. I was worried if users were going to evaluate long strings and not get the full string.

Reply:
I guess. Though if the string is a megabyte, you probably don't want that either :). Doesn't seem like there is a great answer, so pick whichever I guess.

Reply:
I agree with you. I think we will assume the majority of printed strings will be 200 or less. If users want to see more, they can add `set print elements 0` in `setupCommands`.